boston-housing/data.js

```diff
@@ -15,78 +15,108 @@
  * =============================================================================
  */
 
-import * as tf from '@tensorflow/tfjs';
-import * as utils from './utils';
+const Papa = require('papaparse');
+
 
 // Boston Housing data constants:
-const TRAIN_DATA = 'train-data';
-const TRAIN_TARGET = 'train-target';
-const TEST_DATA = 'test-data';
-const TEST_TARGET = 'test-target';
+const BASE_URL =
+    'https://storage.googleapis.com/tfjs-examples/multivariate-linear-regression/data/';
+
+const TRAIN_FEATURES_FN = 'train-data.csv';
+const TRAIN_TARGET_FN = 'train-target.csv';
+const TEST_FEATURES_FN = 'test-data.csv';
+const TEST_TARGET_FN = 'test-target.csv';
+
+/**
+ *
+ * @param {Array<Object>} data Downloaded data.
+ *
+ * @returns {Promise.Array<number[]>} Resolves to data with values parsed as floats.
+ */
+const parseCsv = async (data) => {
+  return new Promise(resolve => {
+    data = data.map((row) => {
+      return Object.keys(row).map(key => parseFloat(row[key]));
+    });
+    resolve(data);
+  });
+};
+
+/**
+ * Downloads and returns the csv.
+ *
+ * @param {string} filename Name of file to be loaded.
+ *
+ * @returns {Promise.Array<number[]>} Resolves to parsed csv data.
+ */
+export const loadCsv = async (filename) => {
+  return new Promise(resolve => {
+    const url = `${BASE_URL}${filename}`;
+
+    console.log(`  * Downloading data from: ${url}`);
+    Papa.parse(url, {
+      download: true,
+      header: true,
+      complete: (results) => {
+        resolve(parseCsv(results['data']));
+      }
+    })
+  });
+};
+
+
 
 /** Helper class to handle loading training and test data. */
 export class BostonHousingDataset {
   constructor() {
-    this.dataset = null;
+    // Arrays to hold the data.
+    this.trainFeatures = null;
+    this.trainTarget = null;
+    this.testFeatures = null;
+    this.testTarget = null;
+    // Metadata
     this.trainSize = 0;
     this.testSize = 0;
-    this.trainBatchIndex = 0;
-    this.testBatchIndex = 0;
-
-    this.NUM_FEATURES = 12;
   }
 
   get numFeatures() {
-    return this.NUM_FEATURES;
+    return this.trainFeatures[0].length;
   }
 
   /** Loads training and test data. */
   async loadData() {
-    this.dataset = await Promise.all([
-      utils.loadCsv(TRAIN_DATA), utils.loadCsv(TRAIN_TARGET),
-      utils.loadCsv(TEST_DATA), utils.loadCsv(TEST_TARGET)
-    ]);
-
-    let {dataset: trainDataset, vectorMeans, vectorStddevs} =
```

Comment:
I was not liking this part of code myself. Thanks for cleaning it up! :)

Reply:
:)